AlphaWallet/Tokens/Types/TokenCollection.swift

```diff
@@ -15,53 +16,61 @@ protocol TokenAddable {
 
+    func mark(token: TokenIdentifiable, isHidden: Bool)
+    func token(for contract: AlphaWallet.Address) -> Token?
+    func token(for contract: AlphaWallet.Address, server: RPCServer) -> Token?
     func fetch()
+    func start()
 }
 
 ///This contains tokens across multiple-chains
 class MultipleChainsTokenCollection: NSObject, TokenCollection {
-
+    
     let tokensFilter: TokensFilter
-    private var tokensViewModelSubject: CurrentValueSubject<TokensViewModel, Never>
+    private var tokensSubject: CurrentValueSubject<[TokenViewModel], Never> = .init([])
 
     private let refereshSubject = PassthroughSubject<Void, Never>.init()
     private var cancelable = Set<AnyCancellable>()
 
     let tokensDataStore: TokensDataStore & DetectedContractsProvideble
-    var tokensViewModel: AnyPublisher<TokensViewModel, Never> {
-        tokensViewModelSubject.eraseToAnyPublisher()
+    var tokens: AnyPublisher<[TokenViewModel], Never> {
+        tokensSubject.eraseToAnyPublisher()
     }
-    private let queue = DispatchQueue(label: "com.MultipleChainsTokenCollection.updateQueue")
+    private let sessions: ServerDictionary<WalletSession>
+    private let config: Config
 
-    init(tokensFilter: TokensFilter, tokensDataStore: TokensDataStore & DetectedContractsProvideble, config: Config) {
+    init(tokensFilter: TokensFilter, tokensDataStore: TokensDataStore & DetectedContractsProvideble, sessions: ServerDictionary<WalletSession>, config: Config) {
         self.tokensFilter = tokensFilter
         self.tokensDataStore = tokensDataStore
-
-        let enabledServers = config.enabledServers
-        let tokens = tokensDataStore.enabledTokens(for: enabledServers)
-        self.tokensViewModelSubject = .init(.init(tokensFilter: tokensFilter, tokens: tokens, config: config))
+        self.sessions = sessions
+        self.config = config
         super.init()
+    }
 
-        tokensDataStore
-            .enabledTokensPublisher(for: enabledServers)
-            .receive(on: queue)
-            .combineLatest(refereshSubject, { tokens, _ in tokens })
-            .map { MultipleChainsTokensDataStore.functional.erc20AddressForNativeTokenFilter(servers: enabledServers, tokens: $0) }
-            .map { TokensViewModel.functional.filterAwaySpuriousTokens($0) }
-            .map { TokensViewModel(tokensFilter: tokensFilter, tokens: $0, config: config) }
-            .debounce(for: .seconds(Constants.refreshTokensThresholdSec), scheduler: queue)
+    func start() {
+        cancelable.cancellAll()
+
+        let initialOrForceSpanshot = Publishers.Merge(Just<Void>(()), refereshSubject)
+            .map { [tokensDataStore, config] _ in tokensDataStore.enabledTokens(for: config.enabledServers) }
+            .eraseToAnyPublisher()
+
+        let newOrChanged = tokensDataStore.enabledTokensPublisher(for: config.enabledServers)
+            .dropFirst()
+            .debounce(for: .seconds(Constants.refreshTokensThresholdSec), scheduler: RunLoop.main)
```

Comment:
to make publisher return value immediatelly we don't want to handle initial value from publisher, that is why do we use `.dropFirst()` instead of this we gate snapshot and return it as .just(<value>).
`debounce` is needed to throttle updates, as data base could be updated to quiqckly, it could cause screen freeze